File: src/main/java/com/agrifund/entities/Banque.java
package com.agrifund.entities;

/**
 * Entité Banque - hérite de Utilisateur
 * Correspond à la table Banque dans la base de données
 */
public class Banque extends Utilisateur {

    private int banqueId; // ID dans la table Banque
    private int utilisateurId; // Référence vers la table Utilisateur
    private String codeBanque;
    private String addresseSiege;
    private String representantLegal;
    private String adresseAgence;
    private String logo;
    private String siteWeb;
    private String statusCompte;
    private boolean compteVerifie;

    // Constructeurs
    public Banque() {
        super();
        this.statusCompte = "en_attente";
        this.compteVerifie = false;
    }

    public Banque(String nom, String prenom, String email, String password,
                  String codeBanque, String addresseSiege, String representantLegal) {
        super(nom, prenom, email, password);
        this.codeBanque = codeBanque;
        this.addresseSiege = addresseSiege;
        this.representantLegal = representantLegal;
        this.statusCompte = "en_attente";
        this.compteVerifie = false;
    }

    // Getters et Setters
    public int getBanqueId() {
        return banqueId;
    }

    public void setBanqueId(int banqueId) {
        this.banqueId = banqueId;
    }

    public int getUtilisateurId() {
        return utilisateurId;
    }

    public void setUtilisateurId(int utilisateurId) {
        this.utilisateurId = utilisateurId;
    }

    public String getCodeBanque() {
        return codeBanque;
    }

    public void setCodeBanque(String codeBanque) {
        this.codeBanque = codeBanque;
    }

    public String getAddresseSiege() {
        return addresseSiege;
    }

    public void setAddresseSiege(String addresseSiege) {
        this.addresseSiege = addresseSiege;
    }

    public String getRepresentantLegal() {
        return representantLegal;
    }

    public void setRepresentantLegal(String representantLegal) {
        this.representantLegal = representantLegal;
    }

    public String getAdresseAgence() {
        return adresseAgence;
    }

    public void setAdresseAgence(String adresseAgence) {
        this.adresseAgence = adresseAgence;
    }

    public String getLogo() {
        return logo;
    }

    public void setLogo(String logo) {
        this.logo = logo;
    }

    public String getSiteWeb() {
        return siteWeb;
    }

    public void setSiteWeb(String siteWeb) {
        this.siteWeb = siteWeb;
    }

    public String getStatusCompte() {
        return statusCompte;
    }

    public void setStatusCompte(String statusCompte) {
        this.statusCompte = statusCompte;
    }

    public boolean isCompteVerifie() {
        return compteVerifie;
    }

    public void setCompteVerifie(boolean compteVerifie) {
        this.compteVerifie = compteVerifie;
    }

    @Override
    public String toString() {
        return "Banque{" +
                "banqueId=" + banqueId +
                ", codeBanque='" + codeBanque + '\'' +
                ", addresseSiege='" + addresseSiege + '\'' +
                ", representantLegal='" + representantLegal + '\'' +
                ", statusCompte='" + statusCompte + '\'' +
                ", compteVerifie=" + compteVerifie +
                ", " + super.toString() +
                '}';
    }
}

File: src/Entity/Banque.php
<?php

namespace App\Entity;

use App\Repository\BanqueRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: BanqueRepository::class)]
class Banque
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'banqueid')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Utilisateur $utilisateur_id = null;

    #[ORM\Column(length: 50)]
    private ?string $codebanque = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $addresseSiege = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $representantLegal = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $adresseAgence = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $logo = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $siteweb = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $statusCompte = null;

    #[ORM\Column(nullable: true)]
    private ?bool $compteverfiee = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): static
    {
        $this->id = $id;

        return $this;
    }

    public function getUtilisateurId(): ?Utilisateur
    {
        return $this->utilisateur_id;
    }

    public function setUtilisateurId(?Utilisateur $utilisateur_id): static
    {
        $this->utilisateur_id = $utilisateur_id;

        return $this;
    }

    public function getCodebanque(): ?string
    {
        return $this->codebanque;
    }

    public function setCodebanque(string $codebanque): static
    {
        $this->codebanque = $codebanque;

        return $this;
    }

    public function getAddresseSiege(): ?string
    {
        return $this->addresseSiege;
    }

    public function setAddresseSiege(?string $addresseSiege): static
    {
        $this->addresseSiege = $addresseSiege;

        return $this;
    }

    public function getRepresentantLegal(): ?string
    {
        return $this->representantLegal;
    }

    public function setRepresentantLegal(?string $representantLegal): static
    {
        $this->representantLegal = $representantLegal;

        return $this;
    }

    public function getAdresseAgence(): ?string
    {
        return $this->adresseAgence;
    }

    public function setAdresseAgence(?string $adresseAgence): static
    {
        $this->adresseAgence = $adresseAgence;

        return $this;
    }

    public function getLogo(): ?string
    {
        return $this->logo;
    }

    public function setLogo(?string $logo): static
    {
        $this->logo = $logo;

        return $this;
    }

    public function getSiteweb(): ?string
    {
        return $this->siteweb;
    }

    public function setSiteweb(?string $siteweb): static
    {
        $this->siteweb = $siteweb;

        return $this;
    }

    public function getStatusCompte(): ?string
    {
        return $this->statusCompte;
    }

    public function setStatusCompte(?string $statusCompte): static
    {
        $this->statusCompte = $statusCompte;

        return $this;
    }

    public function isCompteverfiee(): ?bool
    {
        return $this->compteverfiee;
    }

    public function setCompteverfiee(?bool $compteverfiee): static
    {
        $this->compteverfiee = $compteverfiee;

        return $this;
    }
}
